cmake/configure/configure_2_boost.cmake

```diff
@@ -63,15 +63,19 @@ MACRO(FEATURE_BOOST_CONFIGURE_COMMON)
   ELSE()
     ADD_FLAGS(CMAKE_REQUIRED_FLAGS "-Werror")
   ENDIF()
+  LIST(APPEND CMAKE_REQUIRED_INCLUDES ${BOOST_INCLUDE_DIRS} ${BOOST_BUNDLED_INCLUDE_DIRS})
```

Comment:
This appears to be necessary but I think we might want to add a comment explaining that the first one will be empty if we use the bundled boost (and, in the opposite circumstance, we will pick headers from the first directory if both are present). It took me a minute to understand this.